testing/selenium_tests.py

```diff
@@ -0,0 +1,625 @@
+
+    def test_failed_logon_and_reason(self):
+        """Is there any data?"""
+        panel = load_panel("Failed logon and reason (status code)")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_failed_logons(self):
+        """Is there any data?"""
+        panel = load_panel("Failed Logons")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_failed_logon_status_codes(self):
+        """Is there any data?"""
+        panel = load_panel("Failed logon status codes")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_log_cleared_event_id_1102_or_104(self):
+        """Is there any data?"""
+        panel = load_panel("Log Cleared - event ID 1102 or 104")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_security_log_events_detail(self):
+        """Is there any data?"""
+        panel = load_panel("Security log events - Detail")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_security_log_process_creation_event_id_4688(self):
+        """Is there any data?"""
+        panel = load_panel("Security log - Process creation - event ID 4688")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_security_log_logon_created_logon_type_2(self):
+        """Is there any data?"""
+        panel = load_panel("Security log - Logon created - Logon type 2")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_security_log_network_logon_created_type_3(self):
+        """Is there any data?"""
+        panel = load_panel("Security log - network logon created - Logon type 3")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_security_log_logon_as_a_service_type_5(self):
+        """Is there any data?"""
+        panel = load_panel("Sercurity log - logon as a service - Logon type 5")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_credential_sent_as_clear_text_type_8(self):
+        """Is there any data?"""
+        panel = load_panel("Security log - Credential sent as clear text - Logon type 8")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_logons_with_special_privileges(self):
+        """Is there any data?"""
+        panel = load_panel("Security log - Logons with special privileges assigned - event ID 4672")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_process_started_with_different_creds(self):
+        """Is there any data?"""
+        panel = load_panel("Security log - Process started with different credentials- event ID 4648 [could be RUNAS, scheduled tasks]")

```

Comment:
Would it be worth refactoring this to have an array with all of the panel titles and then have a function that had those 3 lines in it that iterated through the array?  It would shorten the code.

Reply:
To Linda's comment...I've wondered about that myself. It would greatly reduce the code length. The benefit of doing it the way it is right now that it makes it so that each panel has its own test case. If we did the array method, it would be all or nothing. Either all panels work (test passes) or at least one isn't working right (test fails). Maybe it's possible to make the unittesting library have that level of granularity even if we did the array method, but I'm not sure, I haven't looked into it.

Reply:
Ok, you'd need to play around with it, but the `subtest` might do what we want. Never used it before so I'm not sure though. https://stackoverflow.com/questions/26458634/for-loop-in-unittest

Reply:
I'm not sure why we'd want a whole test to fail if only one piece fails.  It would be quicker to fix if we knew all the failures at one time.  In other test suites, there's the concept of "expect" vs "assert". "expect" fails and then continues.  It looks like Selenium has expect, has anyone tried that?  

I agree that subtests look like something interesting to try.

Reply:
We're on the same page here, we definitely don't want the whole test to fail if only one piece fails.